src/hepattn/models/attention.py

```diff
@@ -199,7 +199,15 @@ def recombine_heads(self, x: Tensor) -> Tensor:
             x = x.transpose(-3, -2)  # B H S Dh -> B S H Dh
         return x.flatten(-2)  # B S H Dh -> B S D
 
-    def _prepare_qkv(self, q: Tensor, kv: Tensor | None = None, initial_values: dict | None = None) -> tuple[Tensor, Tensor, Tensor]:
+    def _prepare_qkv(
+        self,
+        q: Tensor,
+        kv: Tensor | None = None,
+        v: Tensor | None = None,
```

Comment:
Given the changes suggested in forward(), I think we know that kv and v are not None, and we can simplify this accordingly